### DISCUSSION TITLE
How can i create .rec file for R

### DISCUSSION THREAD

**Discussion by jasperDD** [apache/incubator-mxnet#14164] [Q&A]
suppose, as example just take captcha picture from this classic task
https://github.com/apache/incubator-mxnet/tree/master/example/captcha
so, i create train.lst
https://github.com/apache/incubator-mxnet/blob/master/example/captcha/captcha_example.png
my train.lst file contain
0	6	6	4	3	train /C:/Users/Adm/Downloads/pic/captcha_example.png
How to get .rec file for it in R language?

**Comment by Ishitori** [comment]
`.Rec` file is a binary file and should be independent from the platform - once you have it, you should be able to use either R or Python.

You can create both `.lst` file and `.rec` file using `im2rec` tool. Please, look at this explanation how to do it: https://github.com/leocvml/mxnet-im2rec_tutorial

**Comment by jasperDD** [comment]
in R there is im2rec function, 
but  my train.lst contains
0	6	6	4	3	train/C:/Users/adm/Downloads/pic/captcha_example.png

library(“mxnet”)
im2rec(image_lst=“C:/Users/adm/Desktop/train.lst”, root=‘C:/Users/adm/Downloads/captcha_example’,
output_rec=‘C:/Users/adm/Desktop/my.rec’, label_width = 4L, pack_label = 0L,
new_size = -1L, nsplit = 1L, partid = 0L, center_crop = 0L,
quality = 95L, color_mode = 1L, unchanged = 0L, inter_method = 1L,
encoding = “.png”)

then i get the error
Error in mx.internal.im2rec(image_lst, root, output_rec, label_width, :
could not find function “mx.internal.im2rec”

What’s wrong
hoe to correct set the arguments

**Comment by Ishitori** [comment]
Unfortunately, im2rec doesn’t work from R. There is an open issue about that, but so far it hasn’t been done: https://github.com/apache/incubator-mxnet/issues/7273

Please, use python or C++ to generate both .lst and .rec files. Take a look at this tutorial for more information: https://github.com/leocvml/mxnet-im2rec_tutorial

**Comment by jasperDD** [comment]
[Ishitori](/Ishitori) may i ask you help me here?
https://github.com/apache/incubator-mxnet/issues/14204
